### DISCUSSION TITLE
I tried the server in example folder, but I cannot find the ble server on my iphone

### DISCUSSION THREAD

**Discussion by unicorn200066** [kevincar/bless#139] [Q&A]
**Describe the solution you'd like**

**Describe alternatives you've considered**
If applicable, provide a clear and concise description of any alternative solutions or features you've considered.

**Additional context**

**Comment by unicorn200066** [comment]
`import asyncio
import logging
import uuid
from time import sleep
from typing import Optional

from bless import *

logging.basicConfig(level=logging.DEBUG)
logger = logging.getLogger(name=__name__)


async def test_ble_server(loop):
    trigger = asyncio.Event()
    trigger.clear()
    server : BlessServer = BlessServer("MyBleServer", loop=loop)

    service_uuid : str = str(uuid.uuid4())
    await server.add_new_service(service_uuid)


    char_uuid : str = str(uuid.uuid4())
    char_flags: GATTCharacteristicProperties = (
            GATTCharacteristicProperties.read |
            GATTCharacteristicProperties.write |
            GATTCharacteristicProperties.notify
    )
    value: Optional[bytearray] = None
    permissions: GATTAttributePermissions = (
            GATTAttributePermissions.readable |
            GATTAttributePermissions.writeable
    )
    await server.add_new_characteristic(service_uuid, char_uuid, char_flags, value, permissions)

    await server.start()
    logger.info(f"server name is {server.name}")
    is_advertising = await server.is_advertising()
    logger.info(f"server is advertising: {is_advertising}")

    await trigger.wait()

if __name__ == "__main__":
    loop = asyncio.get_event_loop()
    loop.run_until_complete(test_ble_server(loop))

`

I ran the code above, however while the server is running, I could not find "MyBleServer" device on my iphone

**Reply by kevincar** [reply]
Hi @unicorn200066 

> I could not find "MyBleServer" device on my iphone

Can you confirm that you're still unable to find it when searching with the [nRFConnect](https://apps.apple.com/us/app/nrf-connect-for-mobile/id1054362403) or the [LightBlue](https://apps.apple.com/us/app/lightblue/id557428110) apps? Or are you using your own iOS code?

**Reply by unicorn200066** [reply]
Thank you for your reply
yes，I found nothing in nRFConnect app, I wonder if I made a wrong uuid or I forgot something in my code. I run the code with win10 OS  and pyCharm IDE, is there anything I need to fix.

**Reply by unicorn200066** [reply]
Hi kevincar
I checked in nRFConnect again, I found my computer's name in it and the device information is 180A.
I found that the service's uuid is the same with my code set, however, it showed that it's an unknown Service, not the name of "MyBleServer" defined in my code, why is that?
Plus, how can I let my iphone know this device as the name I defined.
Thank you so much!
yida wang

**Reply by kevincar** [reply]
Yes, the method for applying the service name to the device is operating system dependent. In other words, Linux, MacOS, and Win10, all apply the name differently and isn't always broadcast the same. The code was last tested on Win11.
